### ISSUE TITLE
openapi-v3-types module: add basic tests

### ISSUE THREAD

**Issue by bajtos** [strongloop/loopback-next#181]
Add basic tests to verify that example OpenAPI v3 documents and Swagger documents (via converter) can be compiled against the type definition. Create test infrastructure that makes it easy to add focused tests in the future.

This is a follow-up issue for #177.

## Acceptance Criteria
- [x] Add validation test for OpenAPI v3 document
- [ ] Add validation test for Swagger document (converted from v3 document)
- [x] Add scaffolding to allow unit tests in future

## BLOCKED BY

 - #301
 - #753

**Comment by shimks** [comment]
@jannyHou Could you cross off the tasks here that have been done as part of your work on #753?

**Comment by jannyHou** [comment]
@shimks I don't think this is done in #753, we don't have tests for package `loopback-next/openapi-v3-types` and `loopback-next/openapi-spec-builder`. I see a bunch of discussion linked to this issue, and am confused by the description
> Add basic tests to verify that example OpenAPI v3 documents and Swagger documents (via converter) can be compiled against the type definition
Could you summarize a little bit what are we adding in this issue? tests for `openapi-v3-types` or `openapi-spec-builder` or both?

**Comment by shimks** [comment]
> Could you summarize a little bit what are we adding in this issue? tests for `openapi-v3-types` or `openapi-spec-builder` or both?

I think the fact that we need to ask this question is an indication that this story needs re-grooming.
@strongloop/lb-next-dev Can we get a discussion going to get the clarification we need for this topic?

**Comment by dhmlau** [comment]
I guess we need clarification from @bajtos?

**Comment by dhmlau** [comment]
Looking at this issue again, I'm thinking maybe the acceptance criteria is clear enough?  The title of this ticket is `openapi-v3-types module: add basic tests`.  Doesn't it mean adding test for at least `openapi-v3-types`?  

Should we leave adding tests for `openapi-spec-builder` as out-of-scope (thus create new issue to track it) or as part of the acceptance criteria (then we edit the criteria above)?

**Comment by shimks** [comment]
Now that I've taken a look at this issue again, I think it's fine as it is now :p (sorry for the confusion). I personally think adding in tests for `openapi-spec-builder` should be done as a part of the task.

**Comment by dhmlau** [comment]
ok. i'll remove `needs clarification` tag then. thanks.

**Comment by virkt25** [comment]
`Add validation test for Swagger document (converted from v3 document)` is no longer relevant as the we only support OAS 3 only now.
